### ISSUE TITLE
findActivePopup cascading popups not working correctly

### ISSUE THREAD

**Issue by DaveBrad** [joel-costigliola/assertj-swing#184]
BasicRobot.java findActivePopup is not working. This was reported before and recently pulled in (but at the time the source was unavailable for me to get to).

'findActivePopup' finds all the active popups in a related cascade of popups. The findActivePopup code does some filtering on the popups to get components through the popups but is poor. 

The filter in findOuterPopupMenu--> popupMenus looks for JPopupMenu in the components list. There can also be JMenuItems as components which have an automatic popup arrangement.

Unfortunately, the components can be JPopupMenu or JMenuItem. If the cascade is made by the use of JMenuItem then AssertJ-Swing findActiveMenu does not work.(JMenuitem adds a sub-JPopupMenu automatically when the JMenuItem has menu items added to it.)

JPopupMenus are of two kinds: JMenu (menubar stuff) and contextual popup (right-click) in a list, combo, canvas,... The Java tutorial 

https://docs.oracle.com/javase/tutorial/uiswing/components/menu.html shows these two kinds of arrangments. The JMenuItem contextual kind looks like:

I've looked at the code below, but due to lack of comments I'm unable to fathom why the filtering was done so unable to suggest a fix for this issue.

`@RunsInEDT
  @Override
  public @Nullable JPopupMenu findActivePopupMenu() {
    JPopupMenu popup = activePopupMenu();
    if (popup != null || isEventDispatchThread()) {
      return popup;
    }
    TimeoutWatch watch = startWatchWithTimeoutOf(POPUP_TIMEOUT);
    while ((popup = activePopupMenu()) == null) {
      if (watch.isTimeOut()) {
        break;
      }
      pause(100);
    }
    return popup;
  }

@RunsInEDT
  private @Nullable JPopupMenu activePopupMenu() {
    List<Component> found = newArrayList(finder().findAll(POPUP_MATCHER));
    if (found.size() >= 1) {
      return findOuterPopupMenu(found);
    }
    return null;
  }

  @RunsInEDT
  private JPopupMenu findOuterPopupMenu(List<Component> found) {
    if (found.size() == 1) {
      return (JPopupMenu) found.get(0);
    }
    List<JPopupMenu> innerMenus = newArrayList();
    for (Component component : found) {
      innerMenus.addAll(popupMenus(((JPopupMenu) component).getComponents()));
    }
    found.removeAll(innerMenus);
    if (found.size() == 1) {
      return (JPopupMenu) found.get(0);
    }

********** oops this reports multiples for a found;Size of zero (0)
    throw multiplePopupMenusFound(found);
  }.
..
@RunsInEDT
  private Collection<? extends JPopupMenu> popupMenus(Component[] components) {
    Set<JPopupMenu> menus = newHashSet();
     
    for (Component component : components) {
      if (component instanceof JPopupMenu) {
        menus.add((JPopupMenu) component);

****** This filter is a mistake and I do not understand, lack of comments as to purpose ?
      }
    }
    return menus;
  }`

**Comment by croesch** [comment]
Okay thanks for your effort! I got your tests running only by fixing `should_Return_Outer_PopupMenu_When_Having_A_CascadingCascadePopup_Select_N` by adding

```
    // crude but effect way to get the cascade
    Component o = found.getComponent(1);
    robot().focus(o);
```

to select the first menu. Is this what you expect to happen or did I broke your test intention? ;)

Don't worry about code style, will format it again. Also I might reduce the complexity of the code a bit. Working with stack trace elements simply to pass an integer seems to be a bit of overengeneering :)
